### ISSUE TITLE
More generic overloads for `world.` and `entity.` methods

### ISSUE THREAD

**Issue by genaray** [genaray/Arch#20]
Would be probably a cool addition to the existing API to allow some generic methods to have more generic overloads...

```csharp
var entity = world.Create<Health,Transform>();
var entity = world.Create(in Health(), in Transform(), ...);

entity.Set(new Health(), new Transform(),...);
entity.Get<Health, Transform>(out var health, out var transform);
entity.Add<Stats,AI>();
entity.Remove<Stats,AI>();

// The same required for world. since entity. relies on them
world.Set(new Health(), new Transform(),...);
world.Get<Health, Transform>(out var health, out var transform);
world.Add<Stats,AI>();
world.Remove<Stats,AI>();
```

This would reduce boilerplate code and would also speed up some stuff :) 
Probably #19 needed, since the generic overloads would likely need to target the fitting archetype ( atleast for creating, adding and removing ).

**Comment by genaray** [comment]
> If im fine with all of thse, i completely disagree with:
> 
> ```cs
> entity.Get<Health, Transform>(out var health, out var transform);
> world.Get<Health, Transform>(out var health, out var transform);
> ```
> 
> These will create a copy of the components all the time, so you will need to call `ref var k = ref entity.Get<K>()` anyway.
> 
> Also, probably OT, maybe this needs a different issue: operations like `Add/Set` can be merged together imho.

Thanks for the feedback and i totally forgot the copy mechanic in that place :D 

I just wonder if its possible to return multiple references... single entity operations are still slow since each operation needs to search the archetype for acessing its content, would be cool to be able to batch `.Get` operations for the same purpose ( of course you can do the same by just doing `entity.GetArchetype()` to acess the internals yourself but thats not every user friendly ).

```csharp
public ref References<T,...N>{ ref T first, ref T second, ref T third... }
var referencesStruct = entity.Get<...>();
```

That could probably work and speed up things ^^

**Comment by genaray** [comment]
Implemented in [db51cb9](https://github.com/genaray/Arch/commit/db51cb9cffb21aeecc885136015444eb89052089).
There dozends of new generic overloads to ease the development and to batch calls to be more efficient.

Parts of this feature could also act as #21 .

```csharp
var entity = world.Create(new Transform(), new Velocity(),...);
var has = entity.Has<Transform, Velocity>();
var refs = entity.Get<Transform, Velocity>(); // Returns a ref struct with ref fields in it. 
entity.Set(new Transform(), new Velocity());

entity.Add<PowerUp, AI>(new PowerUp(), new AI()); // Changes archetype;
entity.Add<Alive, AdditionalStrength, Flying>();
entity.Remove<Dead, Frozen>();
```

`World`, `Archetype` and `Chunk` also received the same methods and overloads. `Add` and `Remove` are also now more efficient since they can be batched.
